hack/e2e.sh

```diff
@@ -0,0 +1,234 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script replaces hack/ginkgo-e2e.sh when not using a cloud provider.
+#
+# It always runs e2e tests without provider=skeleton and reads cluster info only
+# from KUBECONFIG, with no special GCP setup code and no dependency on cluster/
+#
+# It does not pass any cloud provider related flags and is not compatible with
+# ginkgo-e2e.sh, please use ginkgo-e2e.sh when testing with the existing cloud
+# provider support and tests that require this.
+#
+#
+# Usage: `hack/e2e.sh`.
+#
+# Make sure the current KUBECONFIG points to the desired cluster to test.
+#
+# You also need to build e2e.test and ginkgo, and you should build kubectl though
+# it can fall back to the system kubectl.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+# Options
+################################################################################
+# GINKGO_PARALLEL: set to 'y' to run tests in parallel
+GINKGO_PARALLEL=${GINKGO_PARALLEL:-n}
+# GINKGO_SILENCE_SKIPS: set to 'n' to see S character for each skipped test
+GINKGO_SILENCE_SKIPS=${GINKGO_SILENCE_SKIPS:-y}
+# GINKGO_FORCE_NEWLINES: set to 'y' to print a newline after each S or o character
+GINKGO_FORCE_NEWLINES=${GINKGO_FORCE_NEWLINES:-$( if [ "${CI:-false}" = "true" ]; then echo "y"; else echo "n"; fi )}
+# GINKGO_NO_COLOR: If 'y', Ginkgo's reporter will not use escape sequence to color output.
+#
+# Since Kubernetes 1.25, the default is to use colors only when connected to
+# a terminal. That is the right choice for all Prow jobs (The UI doesn't
+# render them properly).
+GINKGO_NO_COLOR=${GINKGO_NO_COLOR:-$(if [ -t 2 ]; then echo n; else echo y; fi)}
+# E2E_TEST_DEBUG_TOOL: If set, the command executed will be:
+# - `dlv exec` if set to "delve"
+# - `gdb` if set to "gdb"
+# NOTE: for this to work the e2e.test binary has to be compiled with
+# make DBG=1 WHAT=test/e2e/e2e.test
+E2E_TEST_DEBUG_TOOL=${E2E_TEST_DEBUG_TOOL:-}
+# KUBE_DNS_DOMAIN: defaults to cluster.local, passed to tests.
+KUBE_DNS_DOMAIN="${KUBE_DNS_DOMAIN:-cluster.local}"
+# PREPULL_IMAGES: if set, tell e2e.test to pre-pull test images.
+PREPULL_IMAGES="${PREPULL_IMAGES:-false}"
+# KUBECTL: the kubectl binary to use, defaults to the one found in _output
+KUBECTL="${KUBECTL:-"$(kube::util::find-binary "kubectl")"}"
+if [[ ! -x "$KUBECTL" ]]; then
+    echo "WARNING: No kubectl found in _output, falling back on system kubectl ..."
+    KUBECTL="kubectl"
+    if ! command -v "${KUBECTL}"; then
+        echo "ERROR: Failed to find system kubectl fallback, kubectl is required to test. Exiting"
+        exit -1
```

Comment:
```suggestion
        exit 1
```
Follow the Linux utilities' behavior.

Reply:
-1 is 255, there's no real reason it needs to be 1 or -1.

Reply:
We're not taking an action like cp/mv though, the important thing is being non-zero and traceable, I think it makes more sense to keep the consistency within these scripts.

I want to introduce a version with better defaults but make it easy to migrate usage. That should mean dropping lots of now unnecessary overriding defaults (like having to set the "conformance" env to avoid GCE setup behaviors)